scalafix-core/shared/src/main/scala/scalafix/internal/rule/NonUnitStatements.scala

```diff
@@ -0,0 +1,80 @@
+package scalafix.internal.rule
+
+import scala.meta.Term.Block
+import scala.meta._
+import scalafix.SemanticdbIndex
+import scalafix.internal.util.TypeSyntax
+import scalafix.lint.{LintCategory, LintMessage}
+import scalafix.rule.{RuleCtx, RuleName, SemanticRule}
+import scalafix.syntax._
+import scala.meta.contrib.implicits.Equality._
+
+case class NonUnitStatements(index: SemanticdbIndex)
+    extends SemanticRule(index, RuleName("NonUnitStatements")) {
+
+  override def description: String =
+    "This rule ensures that statements should only return Unit"
+
+  private lazy val errorCategory: LintCategory =
+    LintCategory.error("Statements should only return Unit")
+
+  def litType(l: Lit): Option[Type] = l match {
+    case Lit.Int(_) => Some(t"_root_.scala.Int")
+    case Lit.Unit() => Some(t"_root_.scala.Unit")
+    case Lit.Float(_) => Some(t"_root_.scala.Float")
+    case Lit.Double(_) => Some(t"_root_.scala.Double")
+    case Lit.Symbol(_) => Some(t"_root_.scala.Symbol")
+    case Lit.Boolean(_) => Some(t"_root_.scala.Boolean")
+    case Lit.Byte(_) => Some(t"_root_.scala.Byte")
+    case Lit.Char(_) => Some(t"_root_.scala.Char")
+    case Lit.Long(_) => Some(t"_root_.scala.Long")
+    case Lit.Null() => Some(t"_root_.scala.Null")
+    case Lit.String(_) => Some(t"_root_.scala.String")
+    case Lit.Short(_) => Some(t"_root_.scala.Short")
+    case _ => None
+  }
+
+  def statementType(stat: Stat): Option[Type] = stat match {
+    case l: Lit => litType(l)
+    case name: Term.Name => name.symbol.flatMap(_.resultType)
+    case Term.Select(_, name) => statementType(name)
+    case Term.Apply(fun, _) => statementType(fun)
+    case Term.ApplyInfix(_, op, _, _) => statementType(op)
+    case Term.ApplyUnary(op, _) => statementType(op)
```

Comment:
could this be simply `case term: Term` ?

Reply:
What do you mean? How I would get `Term.Name` from abstract `term` in that case?

Reply:
https://github.com/scalacenter/scastie/blob/master/instrumentation/src/main/scala/com.olegych.scastie.instrumentation/Instument.scala#L110

Reply:
nevermind, you are right.